tests/components/application-connector/docs/compass-runtime-agent-tests.md

```diff
@@ -0,0 +1,135 @@
+# Compass Runtime Agent
+
+**Table of Contents**
+
+- [Design and architecture](#design-and-architecture)
+- [Building](#building)
+- [Running](#running)
+  - [Deploy a Kyma cluster locally](#deploy-a-kyma-cluster-locally)
+  - [Test setup](#test-setup---compass-runtime-agent-configuration)
+  - [Run the tests](#run-the-tests)
+- [Debugging](#debugging)
+  - [Running without cleanup](#running-without-cleanup)
+
+## Design and architecture
+
+The tests consist of:
+- [Test resources](../resources/charts/compass-runtime-agent-test/) used to perform the test
+- [Test runner](../test/application-connectivity-validator/) with all the test cases
+
+The tests are executed as a Kubernetes Job on a Kyma cluster where the tested Compass Runtime Agent is installed. The test Job is deployed in the `test` Namespace.
+
+![Compass Runtime Agent tests architecture](assets/compass-runtime-agent-tests-architecture.svg)
+
+The interactions between components are the following:
+
+1. Compass Runtime Agent periodically fetches certificates from Compass Connector.
+2. Compass Runtime Agent periodically fetches applications from Compass Director.
+3. Compass Runtime Agent Test sends GraphQL mutations to Compass Director to create, modify, or delete Applications.
+4. Compass Runtime Agent Test verifies whether corresponding Application CRs were created, modified, or deleted.
+5. Compass Runtime Agent Test verifies whether secret with certificates used for communication with Director was created.
+6. Compass Runtime Agent Test verifies whether secret with CA root certificate used by Istio Gateway was created.
```

Comment:
```suggestion
6. Compass Runtime Agent Test verifies whether Secret with CA root certificate used by Istio Gateway was created.
```

Reply:
I added the definite articles to the suggestion :)